src/site/content/en/notifications/use-push-notifications-to-engage-users/index.md

```diff
@@ -0,0 +1,161 @@
+
+The real power of notifications comes from the combination of service workers and push technology.
+
+*   Service workers can run in the background and display notifications even when your app itself is not visible on screen. 
+
+*   Push technology lets a server decide when to send notifications. A push service creates unique URLs for each subscribed service worker. Sending messages to the URL raises events on the service worker.
+
+In the following example flow, the client and service worker use vanilla JavaScript with no extra libraries. The server is built with the [`express` npm package](https://www.npmjs.com/package/express) on [Node.js](https://nodejs.org/en/), and uses the [`web-push` npm package](https://www.npmjs.com/package/web-push) to send notifications. To send information to the server, the client makes a call to a POST URL that the server has exposed.
+
+### Part 1: Registration and subscription
+
+1.  A client app registers a service worker with `ServiceWorkerContainer.register()`. The registered service worker will continue to run in the background when the client is inactive. 
+
+    Client code: 
+
+    ```js
+    navigator.serviceWorker.register('sw.js');
+    ```
+
+1.  The client requests permission from the user to send notifications.
+
+    Client code: 
+
+    ```js
+    Notification.requestPermission();
+    ```
+
+1.  To enable push notifications, the service worker uses `PushManager.subscribe()` to create a push subscription. In the call to `PushManager.subscribe()`, the service worker supplies the app's API key as an identifier.
+
+    Client code:
+
+    ```js/0-1/0/
+    navigator.serviceWorker.register('sw.js').then(sw => {
+      sw.pushManager.subscribe({ /* API key */ });
+    });
+    ```
+
+    Push services like Firebase Cloud Messaging operate on a subscription model. When a service worker subscribes to a push service by calling `PushManager.subscribe()`, the push service creates a URL unique to that service worker. The URL is known as a subscription endpoint. 
+
+1.  The client sends the subscription endpoint to the app server. 
+
+    Client code:
+
+    ```js/1-3/1/
+    navigator.serviceWorker.register('sw.js').then(sw => {
+      sw.pushManager.subscribe({ /* API key */ }).then(subscription => {
+        sendToServer(subscription, '/new-subscription', 'POST');
+      });
+    });
+    ```
+
+    Server code: 
+
+    ```js
+    app.post('/new-subscription', (request, response) => {
+      // extract subscription from request
+      // send 'OK' response
+    });
+    ```
+
+Part 2: Send Notification
+
+1.  The web server sends a notification to the subscription endpoint.
+
+    Server code:
+
+    ```js
+    const webpush = require('web-push');
+
+    let options = { /* config info for cloud messaging and API key */ };  
+    let subscription = { /* subscription created in Part 1*/ };
+    let payload = { /* notification */ };
+
+    webpush.sendNotification(sub, payload, options);
```

Comment:
Shouldn't it be `subscription` instead of `sub`?

Reply:
Yes it should; thank you very much